internal/util/diff/pkgdiff.go

```diff
@@ -120,3 +122,29 @@ func pkgSet(pkgPath string) (sets.String, error) {
 	}
 	return pkgFiles, nil
 }
+
+// nonKptfileEquals returns true if contents of two non-Kptfiles are equal
+// since the changes to addmetricsannotation.CNRMMetricsAnnotation is made
+// by kpt, we should not treat it as changes made by user, so delete the annotation
+// before comparing
+func nonKptfileEquals(s1, s2 string) bool {
```

Comment:
The function name doesn't convey the intent clearly. It will be good to have a comparison function that accepts annotations to ignore.

```Go

func compareResources(r1, r2 string, annotationsToIgnore []string) bool {
}
```

Reply:
It is not just resources, we also compare other files as well which are not KRM resources. We can extend this functionality to take annotations/labels or other parameters as input in the future if there is a need. Since we are starting with one annotation it is better to hardcode IMO.